pxe-server/build/Dockerfile

```diff
@@ -8,10 +8,75 @@ RUN apt-get update && \
         dnsmasq=2.90-4~deb12u1 && \
     rm -rf /var/lib/apt/lists/*
 
+# Download sources for dependent packages and create tar archive
+RUN mkdir -p /sources && \
+    # Install dpkg-dev for source package handling
+    apt-get update && \
+    apt-get install -y --no-install-recommends dpkg-dev tar && \
+    # Create source repositories
+    echo "deb-src http://deb.debian.org/debian bookworm main" > /etc/apt/sources.list.d/debian-sources.list && \
+    echo "deb-src http://deb.debian.org/debian-security bookworm-security main" >> /etc/apt/sources.list.d/debian-sources.list && \
+    # Update with --allow-insecure-repositories to avoid signing conflicts
+    apt-get update --allow-insecure-repositories || apt-get update && \
+    # Define package list with versions
+    echo "dnsmasq:2.90-4~deb12u1" > /tmp/target_packages.txt && \
+    echo "dnsmasq-base:2.90-4~deb12u1" >> /tmp/target_packages.txt && \
+    echo "libdbus-1-3:1.14.10-1~deb12u1" >> /tmp/target_packages.txt && \
+    echo "libjansson4:2.14-2" >> /tmp/target_packages.txt && \
+    echo "libmnl0:1.0.4-3" >> /tmp/target_packages.txt && \
+    echo "libnetfilter-conntrack3:1.0.9-3" >> /tmp/target_packages.txt && \
+    echo "libnfnetlink0:1.0.2-2" >> /tmp/target_packages.txt && \
+    echo "libnftables1:1.0.6-2+deb12u2" >> /tmp/target_packages.txt && \
+    echo "libnftnl11:1.2.4-2" >> /tmp/target_packages.txt && \
+    echo "libxtables12:1.8.9-2" >> /tmp/target_packages.txt && \
+    echo "netbase:6.4" >> /tmp/target_packages.txt && \
+    echo "runit-helper:2.15.2" >> /tmp/target_packages.txt && \
+    # Download source for each package
+    cd /sources && \
+    while IFS=: read package version; do \
+        echo "Getting source for: $package ($version)"; \
+        mkdir -p "$package" && cd "$package" && \
+        apt-get source "$package=$version" 2>/dev/null || \
+        apt-get source "$package" 2>/dev/null || \
+        echo "No source available for $package"; \
+        cd /sources; \
+    done < /tmp/target_packages.txt && \
+    # Install xz-utils for better compression
+    apt-get install -y --no-install-recommends xz-utils && \
+    echo "Creating compressed tar archive of all sources" && \
+    tar -cJf /sources.tar.xz -C /sources . && \
+    # Clean up apt cache
+    rm -rf /var/lib/apt/lists/*
+
 FROM gcr.io/distroless/base-debian12@sha256:cef75d12148305c54ef5769e6511a5ac3c820f39bf5c8a4fbfd5b76b4b8da843
 
-COPY --from=builder /lib /lib
+# Copy only the required libraries for dnsmasq
+COPY --from=builder /lib/x86_64-linux-gnu/libdbus-1.so.3 /lib/x86_64-linux-gnu/
```

Comment:
Avoid copying libc and linker unless needed:
Copying /lib/x86_64-linux-gnu/libc.so.6 and /lib64/ld-linux-x86-64.so.2 can have compatibility implications. Confirm that the distroless base image does not already provide compatible versions.

Reply:
Skipped the linker. Have let the libc be there to ensure I do not change the way it was working earlier.